python/tidy.py

```diff
@@ -536,30 +554,35 @@ def get_file_list(directory, only_changed_files=False):
         # also check untracked files
         args = ["git", "ls-files", "--others", "--exclude-standard", directory]
         file_list += subprocess.check_output(args)
-        return (os.path.join(".", f) for f in file_list.splitlines())
+        for f in file_list.splitlines():
+            yield os.path.join('.', f)
+    elif exclude_dirs:
+        for root, dirs, files in os.walk(directory, topdown=True):
+            # modify 'dirs' in-place so that we don't do unwanted traversals in excluded directories
+            dirs[:] = [d for d in dirs if not any([os.path.join(root, d).startswith(name) for name in ignored_dirs])]
```

Comment:
`[d for d in dirs if not any(os.path.join(root, d).startswith(name) for name in ignored_dirs)]`